redis-sink/src/main/java/org/springframework/cloud/stream/module/redis/CustomRedisConfiguration.java

```diff
@@ -0,0 +1,64 @@
+package org.springframework.cloud.stream.module.redis;
+
+import java.net.UnknownHostException;
+
+import org.apache.commons.pool2.impl.GenericObjectPool;
+import redis.clients.jedis.Jedis;
+
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingClass;
+import org.springframework.boot.autoconfigure.redis.RedisAutoConfiguration;
+import org.springframework.boot.autoconfigure.redis.RedisProperties;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.data.redis.connection.jedis.JedisConnection;
+import org.springframework.data.redis.connection.jedis.JedisConnectionFactory;
+import org.springframework.data.redis.core.RedisOperations;
+
+/**
+ * Creates a dedicated RedisConnectionFactory (that is, different from the spring-cloud-stream one).
+ *
+ * @author Eric Bottard
+ */
+@Configuration
+@ConditionalOnClass({ JedisConnection.class, RedisOperations.class, Jedis.class })
+@EnableConfigurationProperties
+public class CustomRedisConfiguration extends RedisAutoConfiguration {
```

Comment:
If we are creating a custom section for the redis properties, then I think this is the way.  I would rename the class RedisSinkAutoConfiguration.  I am not sure that just the 'module' prefix is ok in terms of structure - seems odd.  The ConditionalOnClass should probably include something from this sink, e.g a ref to RedisSink to be a good citizen.